modules/lmdbbackend/lmdbbackend.cc

```diff
@@ -693,21 +1286,24 @@ bool LMDBBackend::deleteDomain(const DNSName& domain)
 
   { // Remove metadata
     auto txn = d_tmeta->getRWTransaction();
-    auto range = txn.equal_range<0>(domain);
+    LMDBIDvec ids;
+
+    txn.get_multi<0>(domain, ids);
 
-    for (auto& iter = range.first; iter != range.second; ++iter) {
-      iter.del();
+    for (auto _id : ids) {
+      txn.del(_id);
     }
 
     txn.commit();
   }
 
   { // Remove cryptokeys
     auto txn = d_tkdb->getRWTransaction();
-    auto range = txn.equal_range<0>(domain);
+    LMDBIDvec ids;
+    txn.get_multi<0>(domain, ids);
 
-    for (auto& iter = range.first; iter != range.second; ++iter) {
-      iter.del();
+    for (auto _id : ids) {
```

Comment:
This might benefit from a `&`

Reply:
is that actually beneficial for `uint32_t`?

Reply:
Likely not indeed. But in the general case I like the compiler to optimize that, as type defs might change.